cypress/support/step-definitions/flat-table-steps.js

```diff
@@ -12,118 +12,124 @@ const FIFTH_ELEMENT = 4;
     .and('be.visible');
-  flatTabelCellContent(indexRow).eq(SECOND_ELEMENT).should('have.text', 'business')
+  flatTableBodyCellByPosition(indexRow, SECOND_ELEMENT).should('have.text', 'business')
     .and('be.visible');
-  flatTabelCellContent(indexRow).eq(THIRD_ELEMENT).should('have.text', 'Group1, Group2, Group3')
+  flatTableBodyCellByPosition(indexRow, THIRD_ELEMENT).should('have.text', 'Group1, Group2, Group3')
     .and('be.visible');
-  flatTabelCellContent(indexRow).eq(FOURTH_ELEMENT).should('have.text', 'Accounting')
+  flatTableBodyCellByPosition(indexRow, FOURTH_ELEMENT).should('have.text', 'Accounting')
     .and('be.visible');
-  flatTabelCellContent(indexRow).eq(FIFTH_ELEMENT).should('have.text', '12/12/20')
+  flatTableBodyCellByPosition(indexRow, FIFTH_ELEMENT).should('have.text', '12/12/20')
     .and('be.visible');
-  flatTabelCellContent(indexRow).eq(SIXTH_ELEMENT).should('have.text', '20/12/20')
+  flatTableBodyCellByPosition(indexRow, SIXTH_ELEMENT).should('have.text', '20/12/20')
     .and('be.visible');
-  flatTabelCellContent(indexRow).eq(SEVENTH_ELEMENT).should('have.text', '25/12/20')
+  flatTableBodyCellByPosition(indexRow, SEVENTH_ELEMENT).should('have.text', '25/12/20')
     .and('be.visible');
 });
 
-Then('I scroll tabel content to right bottom', () => {
+Then('I scroll table content to right bottom', () => {
   cy.viewport(625, 450);
-  flatTable().parent().scrollTo('100%', '100%');
+  flatTableNoiFrame().parent().scrollTo('100%', '100%');
 });
 
-Then('{string} header cell is not visible', (position) => {
-  switch (position) {
-    case 'second':
-      flatTableHeaderCell(SECOND_ELEMENT).and('not.be.visible');
-      break;
-    case 'third':
-      flatTableHeaderCell(THIRD_ELEMENT).and('not.be.visible');
-      break;
-    case 'fourth':
-      flatTableHeaderCell(FOURTH_ELEMENT).and('not.be.visible');
-      break;
-    case 'fifth':
-      flatTableHeaderCell(FIFTH_ELEMENT).and('not.be.visible');
-      break;
-    default: throw new Error('There are only four not visible FlatTable cells on the page');
+Then('{int} FlatTable rows are {string} visible', (count, state) => {
+  if (state === 'not') {
+    for (let i = 1; i < 2; i++) {
+      flatTableBodyRowByPositionNoiFrame(i).should('not.be.visible');
+    }
+  } else {
+    for (let i = 3; i <= 8; i++) {
+      flatTableBodyRowByPositionNoiFrame(i).should('be.visible');
+    }
   }
 });
 
-Then('{int} FlatTable rows are not visible', (notVisibleRowAmount) => {
-  for (let i = 0; i < notVisibleRowAmount; i++) {
-    flatTableBodyRowByPosition(i).should('not.be.visible');
-  }
+Then('I click on {int} body row', (index) => {
+  flatTableBodyRowByPosition(index).click();
 });
 
-Then('{int} FlatTable rows are visible', (visibleRowAmount) => {
-  for (let i = 2; i <= visibleRowAmount; i++) {
-    flatTableBodyRowByPosition(i).should('be.visible');
-  }
+Then('I focus {int} row and focused row element has golden border on focus', (index) => {
+  cy.wait(500);
```

Comment:
add please DEBUG log flag here and comment why we are adding that `wait`

Reply:
fixed